pulsar-client/src/main/java/org/apache/pulsar/client/impl/PulsarClientImpl.java

```diff
@@ -145,6 +147,8 @@ public PulsarClientImpl(ClientConfigurationData conf, EventLoopGroup eventLoopGr
         conf.getAuthentication().start();
         this.cnxPool = cnxPool;
         externalExecutorProvider = new ExecutorProvider(conf.getNumListenerThreads(), getThreadFactory("pulsar-external-listener"));
+        ioExecutorService = new ThreadPoolExecutor(conf.getNumIoThreads(), conf.getNumIoThreads(), 0L, TimeUnit.MILLISECONDS,
+                new LinkedBlockingQueue<>(), getThreadFactory("pulsar-io-thread"));
```

Comment:
Now we already have threads named `pulsar-client-io`, so it's better to use a different name for the new threads. And I noticed some internal tasks of the client also uses the `externalExecutorProvider` for executing the internal task of the Pulsar client. Maybe we can add an `internalExecutorService` for handing the internal tasks of the client.